File: CPP/Leetcode/125-valid-palindrome/125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int n=s.length();
        string k="";
        transform(s.begin(), s.end(), s.begin(), ::tolower);
        for(int i=0;i<n;i++)
        {
            if((s[i]>='a' && s[i]<='z') || (s[i]>='0' && s[i]<='9')) k+=s[i];
        }
        
        n=k.length();
        int l=0;
        int r=n-1;
        while(l<r)   
        {
            if(k[l]!=k[r]) return false;
            l++;
            r--;
        }
        return true;
    }
};

File: Some Leetcode and GFG Solutions/125-valid-palindrome/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        # st=[i for i in s.lower() if i.isalnum()]
        # return st==st[::-1]
        i=0
        j=len(s)-1
        while i<=j:
            x=s[i].lower()
            y=s[j].lower()
            if x.isalnum() and y.isalnum():
                if x!=y:
                    return False
                else:
                    i+=1
                    j-=1
                    continue
            if not x.isalnum():
                i+=1
            if not y.isalnum():
                j-=1
                
        return True